vk_mem_alloc: Update to latest upstream + volk: Finish update + Adapt custom approach to small objects

RandomShaper opened the pull request:
This updates VMA and instead of using the custom small pool approach from 4e6c9d3ae979f2eb0151cf581fe61d2f3194ea72, lazily creates pools for the relevant memory type indices, which doesn't require patching VMA.

VMA now works with a TLSF algorithm by default, in contrast to the former default (linear) we were using (~10%). This is the new recommended default approach; slightly slower, but allows to free memory instead of growing until exhaustion (see https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/issues/178#issuecomment-1048201561).

Also, Volk is upgraded to the correct version (it seems some commit was supposed to do it, but failed for some reason).

Also, patches already merged upstream or not needed any longer are removed. ~The nullability patch has been upstreamed in https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/pull/242.~ (Already merged.)

Supersedes #58137.

**UPDATE:** Author of VMA thinks this approach is correct (see https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/issues/178#issuecomment-1050171625).

RandomShaper committed:
Really update volk to 1.3.204

akien-mga commented:
> Also, Volk is upgraded to the correct version (it seems some commit was supposed to do it, but failed for some reason).

Indeed, I had updated all related components and eventually forgot to update Volk itself it seems. (Ref #57980)

RandomShaper committed:
vk_mem_alloc: Update to upstream + Adapt approach to small objects pooling

This updates VMA and instead of using the custom small pool approach from 4e6c9d3ae979f2eb0151cf581fe61d2f3194ea72, lazily creates pools for the relevant memory type indices, which doesn't require patching VMA.

Also, patches already merged upstream or not needed any longer are removed.

akien-mga commented:
Thanks!
